hierarchicalforecast/core.py

```diff
@@ -90,170 +113,289 @@ class HierarchicalReconciliation:
+        Most reconciliation methods can be described by the following convenient
         linear algebra notation:
 
         $$\\tilde{\mathbf{y}}_{[a,b],\\tau} = \mathbf{S}_{[a,b][b]} \mathbf{P}_{[b][a,b]} \hat{\mathbf{y}}_{[a,b],\\tau}$$
 
         where $a, b$ represent the aggregate and bottom levels, $\mathbf{S}_{[a,b][b]}$ contains
-        the hierarchical aggregation constraints, and $\mathbf{P}_{[b][a,b]}$ varies across 
-        reconciliation methods. The reconciled predictions are $\\tilde{\mathbf{y}}_{[a,b],\\tau}$, and the 
+        the hierarchical aggregation constraints, and $\mathbf{P}_{[b][a,b]}$ varies across
+        reconciliation methods. The reconciled predictions are $\\tilde{\mathbf{y}}_{[a,b],\\tau}$, and the
         base predictions $\hat{\mathbf{y}}_{[a,b],\\tau}$.
 
         **Parameters:**<br>
-        `Y_hat_df`: pd.DataFrame, base forecasts with columns `ds` and models to reconcile indexed by `unique_id`.<br>
-        `Y_df`: pd.DataFrame, training set of base time series with columns `['ds', 'y']` indexed by `unique_id`.<br>
+        `Y_hat_df`: DataFrame, base forecasts with columns ['unique_id', 'ds'] and models to reconcile.<br>
+        `Y_df`: DataFrame, training set of base time series with columns `['unique_id', 'ds', 'y']`.<br>
         If a class of `self.reconciles` receives `y_hat_insample`, `Y_df` must include them as columns.<br>
-        `S`: pd.DataFrame with summing matrix of size `(base, bottom)`, see [aggregate method](https://nixtla.github.io/hierarchicalforecast/utils.html#aggregate).<br>
+        `S`: DataFrame with summing matrix of size `(base, bottom)`, see [aggregate method](https://nixtla.github.io/hierarchicalforecast/utils.html#aggregate).<br>
         `tags`: Each key is a level and its value contains tags associated to that level.<br>
         `level`: positive float list [0,100), confidence levels for prediction intervals.<br>
         `intervals_method`: str, method used to calculate prediction intervals, one of `normality`, `bootstrap`, `permbu`.<br>
         `num_samples`: int=-1, if positive return that many probabilistic coherent samples.
         `seed`: int=0, random seed for numpy generator's replicability.<br>
-        `sort_df` : bool (default=True), if True, sort `df` by [`unique_id`,`ds`].<br>
+        `sort_df` : deprecated.<br>
         `is_balanced`: bool=False, wether `Y_df` is balanced, set it to True to speed things up if `Y_df` is balanced.<br>
+        `id_col` : str='unique_id', column that identifies each serie.<br>
+        `time_col` : str='ds', column that identifies each timestep, its values can be timestamps or integers.<br>
+        `target_col` : str='y', column that contains the target.<br>
 
         **Returns:**<br>
-        `Y_tilde_df`: pd.DataFrame, with reconciled predictions.
+        `Y_tilde_df`: DataFrame, with reconciled predictions.
         """
+        # Raise deprecation warning for sort_df
+        _maybe_warn_sort_df(sort_df)
```

Comment:
We could also remove this as part of the breaking changes

Reply:
Yeah, maybe better indeed to deprecate all these things in one go